### PR TITLE
Mrs/ci android libnodegl

### PR DESCRIPTION
**Pull request by mrobertseidowsky-gpsw**
### Description:

This PR allows to:
-  Compile **ffmpeg** sources for different Android host,
-  Compile [sxplayer](https://github.com/Stupeflix/sxplayer) sources for different Android host,
-  Build **libnodegl** lib for each target,
- Libnodegl **tests** are compiled but not runned (skipped).

One cross compilation file _(.ini)_ per architecture (_i.e_ processor type on the phone) is given as **meson** input.

Architectures are:
- **x86_64**: x86 processor 64-bit,
- **arm**: arm processor 32-bit,
- **aarch64**: arm processor 64-bit.

 A compiler for each architecture is directly provided from the  **Android ndk toolchain** on the remote machine.

### Notes:

- **meson** is installed with **pip** to receive the most up-to-date version (**apt** provides version `0.53.2` which is not compatible with project requirement (`>=0.57.0`) while **pip** provides version `0.62.0`).
- About the **dependencies**, `sxplayer + ffmpeg` are being rebuilt at the moment, but a solution that catch artifacts dropped on sxplayer repository may be considered in the future.
- The **files tree** has been reshaped because:
    - by default meson set `/usr/local` as prefix in the `.pc` file which can not be written on github action remote machine (only write access to home).
    - by default the generated files has this structure _(tree pruned with only relevant files)_:
    
```
builddir
├── libsxplayer.so
├── meson-private
│   ├── libsxplayer.pc
├── sxplayer.h
```

- So a more standard tree is created in the home thanks to the command `meson install`.
    - prefix `/usr/local` becomes `/home/runner/usr/local`
    - structure of files becomes:

```
/home/runner
├── usr
    ├── local
        ├── lib
            ├── libsxplayer.so
	    ├── pkgconfig
 	        ├── libsxplayer.pc
        ├── include
            ├── sxplayer.h
```
- This structure is, at the end easily understood by meson when building libnodegl.

### CODE REVIEW

File: .github/meson-android-x86_64.ini [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[binaries]
+c = '/usr/local/lib/android/sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/bin/x86_64-linux-android28-clang'
+cpp = '/usr/local/lib/android/sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/bin/x86_64-linux-android28-clang++'
+strip = '/usr/local/lib/android/sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/bin/x86_64-linux-android-strip'
+
```
Comment (mbouron-gpsw) [inline .github/meson-android-x86_64.ini:5]: Extra newline.

File: .github/workflows/ci_android.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        PKG_CONFIG_LIBDIR=$HOME/ffmpeg-install-${{ matrix.arch }}/lib/pkgconfig/ \
+        meson setup --prefix $HOME/usr/local --cross-file .github/meson-android-${{ matrix.arch }}.ini builddir
+        meson install -C builddir
+        meson compile -C builddir -v
+
+    - name: Build nodegl lib for Android
+      run: |
+        cd libnodegl
+        PKG_CONFIG_LIBDIR=$HOME/ffmpeg-install-${{ matrix.arch }}/lib/pkgconfig/:$HOME/usr/local/lib/pkgconfig/ \
+        meson setup --cross-file ../.github/meson-android-${{ matrix.arch }}.ini builddir
```
Comment (mbouron-gpsw) [inline .github/workflows/ci_android.yml:70]: The current rule does not actually compile libnodegl (only the standalone tests), please add `meson compile -C builddir` before `meson test -C builddir -v`

File: .github/workflows/ci_android.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        make install -j$(($(nproc)+1))
+
+    - name: Build sxplayer lib for Android & reshape files tree
+      run: |
+        git clone --depth 1 https://github.com/Stupeflix/sxplayer
+        cd sxplayer
+        PKG_CONFIG_LIBDIR=$HOME/ffmpeg-install-${{ matrix.arch }}/lib/pkgconfig/ \
+        meson setup --prefix $HOME/usr/local --cross-file .github/meson-android-${{ matrix.arch }}.ini builddir
+        meson install -C builddir
+        meson compile -C builddir -v
```
Comment (mbouron-gpsw) [inline .github/workflows/ci_android.yml:64]:
```
        meson install -C builddir
        meson compile -C builddir -v
```
should be the other way around.

File: .github/meson-android-x86_64.ini (hunk 1 above)
Comment (mrobertseidowsky-gpsw) [reply .github/meson-android-x86_64.ini]: removed

File: .github/workflows/ci_android.yml (hunk 2 above)
Comment (mrobertseidowsky-gpsw) [reply .github/workflows/ci_android.yml:70]: done

File: .github/workflows/ci_android.yml (hunk 3 above)
Comment (mrobertseidowsky-gpsw) [reply .github/workflows/ci_android.yml:63]: lines order changed. `install` is necessary (see PR description about files tree).

File: .github/workflows/ci_android.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        meson setup --prefix $HOME/usr/local --cross-file .github/meson-android-${{ matrix.arch }}.ini builddir
+        meson compile -C builddir -v
+        meson install -C builddir
+
+    - name: Build nodegl lib for Android
+      run: |
+        cd libnodegl
+        PKG_CONFIG_LIBDIR=$HOME/ffmpeg-install-${{ matrix.arch }}/lib/pkgconfig/:$HOME/usr/local/lib/pkgconfig/ \
+        meson setup --cross-file ../.github/meson-android-${{ matrix.arch }}.ini builddir
+        meson compile -C builddir
```
Comment (mbouron-gpsw) [inline .github/workflows/ci_android.yml:71]: `meson compile -C builddir -v`

File: .github/workflows/ci_android.yml (hunk 3 above)
Comment (mbouron-gpsw) [reply .github/workflows/ci_android.yml:63]: Thanks, is the -v necessary ? (it looks like it's not used by make test for the desktop platform)

Comment (mrobertseidowsky-gpsw) [reply .github/workflows/ci_android.yml]: updating after our conversation.

**Pull request merged**